qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1860,6 +1859,17 @@ def __json_encode__(self):
 
         # Handle non-serializable objects
         json_value["_jobs"] = self._safe_serialize_jobs()
+        service_value = getattr(self, "_service")
+        if service_value:
+            # the attribute self._service in charge of the connection and communication with the
+            #  experiment db. It doesn't have meaning in the json format so there is no need to serialize
+            #  it.
+            LOG.warning(
```

Comment:
Sound good. I will change it accordingly.
Regarding the location of the warning, I thought that the message will appear in the encode so that the user would not find out (by surprise) that the service was not serialized.

Should we raise it in `__json_decode__` or is it better to raise it in both places?